.github/workflows/merge-main.yml

```diff
@@ -192,46 +197,32 @@ jobs:
       - name: Log in to the Container registry
         uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
         with:
-          registry: ghcr.io
+          registry: ${{ env.REGISTRY }}
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Promoting API
-        uses: shrink/actions-docker-registry-tag@v3
-        with:
-          registry: ghcr.io
-          repository: ${{ github.repository }}/api
-          target: test
-          tags: prod
-
-      - name: Promoting API
-        uses: shrink/actions-docker-registry-tag@v3
-        with:
-          registry: ghcr.io
-          repository: ${{ github.repository }}/api
-          target: test
-          tags: ${{ steps.changelog.outputs.tag }}
-
-      - name: Promoting Init
-        uses: shrink/actions-docker-registry-tag@v3
-        with:
-          registry: ghcr.io
-          repository: ${{ github.repository }}/init
-          target: test
-          tags: prod
-
-      - name: Promoting Init
+  image-promotions:
+    name: Image Promotions
+    needs:
+      - deploy-test
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        component: [api, init]
+        tag: [prod, "${{ needs.deploy-test.outputs.tag }}"]
+    steps:
+      - name: Promoting ${{ matrix.component }}
         uses: shrink/actions-docker-registry-tag@v3
         with:
```

Comment:
I would suggest adding the following to prevent retag issues

```
 if: ${{ matrix.tag != '' }}
continue-on-error: true       
```

The tag validation is just to prevent empty tags (in case of a release skip) breaking the execution flow. The continue-on-error is just a last resort